crates/matrix-sdk/src/sliding_sync/client.rs

```diff
@@ -289,11 +289,19 @@ impl<'a> SlidingSyncResponseProcessor<'a> {
         // `handle_room_response` before this function), so panic is fine.
         assert!(self.response.is_none());
 
-        self.to_device_events =
-            self.client.base_client().process_sliding_sync_e2ee(extensions).await?;
+        self.to_device_events = if let Some(to_device_events) = self
+            .client
+            .base_client()
+            .process_sliding_sync_e2ee(extensions.to_device.as_ref(), &extensions.e2ee)
+            .await?
+        {
+            // Some new keys might have been received, so trigger a backup if needed.
+            self.client.encryption().backups().maybe_trigger_backup();
 
-        // Some new keys might have been received, so trigger a backup if needed.
-        self.client.encryption().backups().maybe_trigger_backup();
```

Comment:
@poljar ought to be able to confirm, but: if you haven't received any e2ee/to-device data, I suppose there's no new key coming from another way, right?

Reply:
There are a bunch of places where room keys can come from, but in a `/sync` response, only the to-device part will carry any room keys.

This would be another place and we call the `maybe_trigger_backup()` method as well:

https://github.com/matrix-org/matrix-rust-sdk/blob/59d3608c32c94b7bf210a9bee1ba2733d37b2cec/crates/matrix-sdk/src/encryption/mod.rs#L1490-L1492